### PR TITLE
Implement the zeitwerk autoloader within lib/msf/base

### PR DESCRIPTION
**Pull request by dwelch-r7**
Continuation of the zeitwerk initiative we have going on, following on from this pr ##14202

In this PR we have zeitwerked the `lib/msf/base` folder, this was significantly easier than the previous work and is hopefully a sign that things will become easier as we go through the rest of the project.

# Verification steps

- [x] All framework tests pass
- [x] Metasploit Pro tests pass

On Linux
- [ ] Console boots up
- [ ] `search` for and `use` a module
- [ ] Sucessfully run a module
- [ ] Generate a payload with `msfvenom`
- [ ] Gain a meterpreter session

On Osx
- [ ] Console boots up
- [ ] `search` for and `use` a module
- [ ] Sucessfully run a module
- [ ] Generate a payload with `msfvenom`
- [ ] Gain a meterpreter session

On Windows
- [ ] Console boots up
- [ ] `search` for and `use` a module
- [ ] Sucessfully run a module
- [ ] Generate a payload with `msfvenom`
- [ ] Gain a meterpreter session

### CODE REVIEW

File: lib/msf/base/sessions/command_shell_unix.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    def initialize(*args)
+      self.platform = "unix"
+      super
+    end
+    def shell_command_token(cmd,timeout = 10)
+      shell_command_token_unix(cmd,timeout)
+    end
+  end
+
+end
```
Comment (adfoster-r7) [inline lib/msf/base/sessions/command_shell_unix.rb:13]: > No newline at the end of file

Might be worth modifying your editor to add this trailing new line :+1:

File: tools/exploit/egghunter.rb [hunk 2]
```diff
@@ -129,7 +128,6 @@ def run
         list_formats
         return
       end
-
```
Comment (adfoster-r7) [inline tools/exploit/egghunter.rb:132]: I'm not strongly against this line being deleted, but if it's just a hang over from deleting previous debug code - it's probably worth reverting

File: lib/msf/base/sessions/command_shell_unix.rb (hunk 1 above)
Comment (dwelch-r7) [reply lib/msf/base/sessions/command_shell_unix.rb:13]: Argh :facepalm:

File: tools/exploit/egghunter.rb (hunk 2 above)
Comment (dwelch-r7) [reply tools/exploit/egghunter.rb:132]: I'll make sure to do that

**Comment by adfoster-r7** [comment]
@dwelch-r7 This seems pretty low risk to me; Have you verified this works on windows as a sanity check?
I know that's something we missed out on the last iteration when we ran into https://github.com/rapid7/metasploit-framework/pull/14514

It might be good to add the PR template's verification steps to ensure nothing's missed :+1:

File: lib/msf/base.rb [hunk 3]
```diff
@@ -16,22 +16,14 @@
 # framework-base depends on framework-core
 
 # Configuration
-require 'msf/base/config'
-require 'msf/base/logging'
 
 # Simple wrapper
-require 'msf/base/simple'
 
 # Sessions
```
Comment (adfoster-r7) [inline lib/msf/base.rb:22]: I'm guessing these comments can be removed now? 👀

File: lib/rex/post/meterpreter/ui/console/command_dispatcher.rb [hunk 4]
```diff
@@ -78,7 +78,6 @@ def msf_loaded?
     if client.framework
       # We have a framework instance so the msf libraries should be
       # available.  Load up the ones we're going to use
-      require 'msf/base/serializer/readable_text'
     end
```
Comment (adfoster-r7) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher.rb:81]: Is this dead now?

File: lib/rex/post/hwbridge/ui/console/command_dispatcher.rb [hunk 5]
```diff
@@ -61,7 +61,6 @@ def msf_loaded?
     if client.framework
       # We have a framework instance so the msf libraries should be
       # available.  Load up the ones we're going to use
-      require 'msf/base/serializer/readable_text'
     end
```
Comment (adfoster-r7) [inline lib/rex/post/hwbridge/ui/console/command_dispatcher.rb:64]: Is this `if` statement dead now?

Comment (dwelch-r7) [reply lib/rex/post/hwbridge/ui/console/command_dispatcher.rb:64]: sure looks it, will remove

File: lib/rex/post/meterpreter/ui/console/command_dispatcher.rb (hunk 4 above)
Comment (dwelch-r7) [reply lib/rex/post/meterpreter/ui/console/command_dispatcher.rb:81]: sure looks it, will remove

File: lib/msf/base.rb (hunk 3 above)
Comment (dwelch-r7) [reply lib/msf/base.rb:22]: I actually have this file being completely removed in it's entirety an upcoming PR, it's no longer necessary but I can clean it up real quick for this one

**Pull request merged**